### PR TITLE
Let entries with the same timestamp stay in the same chunk.

### PR DESCRIPTION
**Pull request by cyulysses-corp**

### CODE REVIEW

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java [hunk 1]
```diff
@@ -109,28 +110,32 @@ private void executeScriptChunk(
       String scriptName,
       Integer chunkNumber)
       throws SQLException, IOException {
-    String firstRowStamp = getUtcTimeStringFromTimestamp(resultSet.getTimestamp(labelColumn));
+    Timestamp previousTimestamp = new Timestamp(0);
```
Comment (frsv) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java:113]: We can not rely on timestamps for split rows. Documentation doesn't guarantee that https://docs.teradata.com/r/Teradata-Database-Data-Dictionary/June-2017/Views-Reference/QryLogSQLV – it says `when DBQL SQL cache was written`. 
Here is an example of such SQL log with different timestamps for the same query ID:

```sql
SELECT CollectTimeStamp, QueryID FROM QryLogSQLV WHERE QueryID = 3071939516
48781243;

 *** Query completed. 2 rows found. 2 columns returned.
 *** Total elapsed time was 1 second.

          CollectTimeStamp               QueryID
--------------------------  --------------------
2022-02-13 16:36:11.240000    307193951648781243
2022-02-13 16:36:13.760000    307193951648781243
```

Comment (frsv) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java:113]: We could rely on QueryID, but still there is a chance when running with timerange boundaries that one of the logs will fall out of these boundraries...

Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java:113]: I should've explained more that this change matters the most for the resumable/incremental runs. They (will, according to the current design) rely on last-saved-timestamps as the "checkpoint" for new runs, whose beginning timestamp of the new query's time range will be the last-saved-timestamp + 1us (Teradata's timestamp resolution). Thus it's important to make sure rows with the same sorting timestamps do not get separated into different chunks.

**Review by frsv — APPROVED** [review]

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java [hunk 2]
```diff
@@ -99,8 +99,8 @@ private Instant getUtcInstantFromDatetimeAndZone(String dateTime, String zoneId)
       })
   private String dbAddress;
 
-  @Option(names = "--db-user", description = "The user name for the database.")
-  private String dbUserName = "";
+  @Option(names = "--db-user", required = true, description = "The user name for the database.")
+  private String dbUserName;
 
   @Option(
       names = "--db-password",
```
[the remark this answers is not in the document]
Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java:106]: (Sorry should've pushed the rebased commit after the original one is pulled. )

Password gets an interactive mode from https://github.com/google/dwh-assessment-extraction-tool/commit/793ebaa7a2233c4317cc38603c0f680cdb028c50 with arity 0/1, so we don't need to impose additional requirements.

[the remark this answers is not in the document]
Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java]: (Resolved in the based commit).

**Pull request merged**